### ISSUE TITLE
[Feature]: Multiple entry point support

### ISSUE THREAD

**Issue by crup** [remix-run/remix#1020]
### What is the new or updated feature that you are suggesting?

Can we get a support for configuring `entryClientFile` as an array instead of string to support multiple output bundles and serve them conditionally via `entryServerFile`? 

### Why should this feature be included?

The idea behind having multiple client entry points is to support multiple outputs of a single app. 
Let's say, I've an application where I want to serve a totally different website on mobile and I don't want to go with the responsive layout. 
So I can have two entry points for the app:
entry.client.mobile.js
entry.client.desktop.js

Or maybe I want to separate out apps based on URLs like:
http://app.com/in
http://app.com/au
http://app.com/fr

So, these can be three different apps for three different countries. 

Is it doable in Remix? Maybe via playing around with context in RemixServer?

**Comment by tigerabrodi** [comment]
@crup That is an interesting feature actually :thinking: 

So you want the same codebase, but within, two different websites, one for mobile and one for tablet/desktop...?

Seems quite complex, mhm.

**Comment by crup** [comment]
@tigerabrodi 
Yes! If I've that capability, I can spit out bundles for mobile/desktop/amp or any other strategy which works for my business requirements. 
It's not that complex but I want to get the point of view of the Remix team on this feature. 
I've actually done that with Electrode(https://github.com/electrode-io/electrode) and it works really well. 

If there are workarounds to support that feature, I will consider using Remix. I've already lost hopes from Next.js.
